pkg/cache/context.go

```diff
@@ -555,11 +555,17 @@ func (ctx *Context) getOrCreateApplication(pod *v1.Pod) *Application {
 
 	if application, ok := ctx.applications[appId]; ok {
 		return application
-	} else {
-		newApp := NewApplication(appId, utils.GetQueueNameFromPod(pod), ctx.schedulerApi)
-		ctx.applications[appId] = newApp
-		return ctx.applications[appId]
 	}
+	// create the tags for the application
+	// labels or annotations from the pod can be added when needed
+    tags := map[string]string{}
+    tags["namespace"] = pod.Namespace
```

Comment:
Do we need some check for the value of `pod.Namespace` here?
According to the doc
```
Namespace defines the space within each name must be unique. An empty namespace is
equivalent to the "default" namespace, but "default" is the canonical representation.
Not all objects are required to be scoped to a namespace - the value of this field for
those objects will be empty.
```
this might be empty.